12-kubernetes-cluster.tf

```
@@ -136,3 +138,45 @@ resource "azurerm_kubernetes_cluster_node_pool" "additional_node_pools" {
   orchestrator_version  = var.kubernetes_cluster_version
   vnet_subnet_id        = data.azurerm_subnet.aks.id
 }
+
+resource "azurerm_monitor_diagnostic_setting" "kubernetes_cluster_diagnostic_setting" {
+  name                       = "DiagLogAnalytics"
+  count                      = var.monitor_diagnostic_setting ? 1 : 0
+  target_resource_id         = azurerm_kubernetes_cluster.kubernetes_cluster.id
+  log_analytics_workspace_id = var.log_workspace_id
+
+  log {
+    category = "kube-apiserver"
+    enabled  = true
+  }
+
+  log {
+    category = "kube-controller-manager"
+    enabled  = true
+  }
+
+  log {
+    category = "cluster-autoscaler"
+    enabled  = true
+  }
+
+  log {
+    category = "kube-scheduler"
+    enabled  = true
+  }
+
+  log {
+    category = "kube-audit"
```

this is what I was talking about yesterday: https://docs.microsoft.com/en-us/azure/aks/view-control-plane-logs#log-categories

kube-audit isn't really recommended as it has a really high cost
```suggestion
    category = "kube-audit-admin"
```